crates/bevy_ecs/src/query/state.rs

```diff
@@ -83,10 +86,47 @@ pub struct QueryState<D: QueryData, F: QueryFilter = ()> {
     pub(super) is_dense: bool,
     pub(crate) fetch_state: D::State,
     pub(crate) filter_state: F::State,
+    sync_by_observer: bool,
     #[cfg(feature = "trace")]
     par_iter_span: Span,
 }
 
+impl<D: QueryData + 'static, F: QueryFilter + 'static> Component for QueryState<D, F> {
+    const STORAGE_TYPE: StorageType = StorageType::SparseSet;
```

Comment:
Why `SparseSet` and not the default of `Table`?

Reply:
Because of the D, F pair, virtually zero queries have the same type, so making it a Table doesn't make much sense.

This was what I was thinking when I first gave it a shot.

Reply:
I switch to table storage now, it's possible to have multiple queries that have the same type even when we do dedup.